tools/clang/lib/SPIRV/ModuleBuilder.cpp

```diff
@@ -474,6 +477,55 @@ uint32_t ModuleBuilder::getFunctionType(uint32_t returnType,
   return typeId;
 }
 
+uint32_t ModuleBuilder::getImageType(uint32_t sampledType, spv::Dim dim,
+                                     bool isArray) {
+  const Type *type = Type::getImage(theContext, sampledType, dim,
+                                    /*depth*/ 0, isArray, /*ms*/ 0,
```

Comment:
`TextureXX`s should be mapped to normal images AFAIK. (Sorry my understanding of how to translating these features is also incomplete right now. I'm not sure how you can express a depth image in HLSL source code.) Glslang does the same thing. Similarly for single-sampled content. Basically copying the behavior of Glslang here. We can always fix / grow new cases later.